lib/inspec/reporters/base.rb

```diff
@@ -5,9 +5,31 @@ class Base
     def initialize(config)
       @config = config
       @run_data = config[:run_data]
+      apply_report_resize_options unless @run_data.nil?
       @output = ""
     end
 
+    # Apply options such as message truncation and removal of backtraces
+    def apply_report_resize_options
+      runtime_config = Inspec::Config.cached.respond_to?(:final_options) ? Inspec::Config.cached.final_options : {}
+
+      message_truncation = runtime_config[:reporter_message_truncation] || "ALL"
+      trunc = message_truncation == "ALL" ? -1 : message_truncation.to_i
+      include_backtrace = runtime_config[:reporter_backtrace_inclusion].nil? ? true : runtime_config[:reporter_backtrace_inclusion]
+
+      @run_data[:profiles]&.each do |p|
```

Comment:
I had a quick look with `benchmark/ips` in the Terminal with an identical data structure to see if I could find another way to make this change without the loops but as far as I could tell this is the best approach!

Reply:
Yes I don't much like these nested loops.
Note the cautious  `&.` because at each level the report might be missing any occurrences of the nested elements.